presto-spark-base/src/main/java/com/facebook/presto/spark/execution/PrestoSparkTaskExecutorFactory.java

```diff
@@ -386,6 +399,42 @@ public PrestoSparkTaskExecutorFactory(
                 allocationTrackingEnabled,
                 false);
 
+        final double memoryRevokingThreshold = getMemoryRevokingThreshold(session);
+        if (isSpillEnabled(session)) {
+            memoryPool.addListener((pool, queryId, totalMemoryReservationBytes) -> {
```

Comment:
`totalMemoryReservationBytes` includes `user memory + system memory + revokable memory` from this query